src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/IValidationAttributeProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using Microsoft.AspNetCore.Mvc.Rendering;
+
+namespace Microsoft.AspNetCore.Mvc.ViewFeatures
+{
+    /// <summary>
+    /// Contract for a service providing validation attributes for expressions.
+    /// </summary>
+    public interface IValidationAttributeProvider
```

Comment:
I'll go w/ `IValidationHtmlAttributeProvider` and leave the `AddValidationAttributes()` method alone unless someone objects.